server/src/main/java/com/eclipsesource/glsp/ecore/gmodel/GModelFactory.java

```diff
@@ -0,0 +1,167 @@
+		if (semanticElement instanceof EClassifier) {
+			result = classifierNodeFactory.create((EClassifier) semanticElement);
+		} else if (semanticElement instanceof EPackage) {
+			result = create((EPackage) semanticElement);
+		} else if (semanticElement instanceof EReference) {
+			result = create((EReference) semanticElement);
+		} else if (semanticElement instanceof ENamedElement) {
+			result = labelFactory.create((ENamedElement) semanticElement);
+		}
+		if (result == null) {
+			throw createFailed(semanticElement);
+		}
+		return result;
+	}
+
+	public GGraph create() {
+		return create(modelState.getEditorContext().getEcoreFacade().getEPackage());
+	}
+
+	public GGraph create(EPackage ePackage) {
+		GGraph graph = getOrCreateRoot();
+		graph.setId(toId(ePackage));
+
+		graph.getChildren().addAll(ePackage.getEClassifiers().stream()//
+				.map(this::create)//
+				.collect(Collectors.toList()));
+
+		graph.getChildren().addAll(ePackage.getEClassifiers().stream() //
+				.filter(EClass.class::isInstance) //
+				.map(EClass.class::cast) //
+				.flatMap(eClass -> createEdges(eClass).stream()) //
+				.collect(Collectors.toList()));
+		return graph;
+
+	}
+
+	private List<GModelElement> createEdges(EClass eClass) {
+		List<GModelElement> children = new ArrayList<>();
+		// create reference edges
+		eClass.getEReferences().stream().map(this::create).forEach(children::add);
+		// create inheritance edges
+		eClass.getESuperTypes().stream().map(s -> create(eClass, s)).forEach(children::add);
+		return children;
+	}
+
+	public GEdge create(EReference eReference) {
+		String label = String.format("[%s..%s] %s", eReference.getLowerBound(),
+				eReference.getUpperBound() == -1 ? "*" : eReference.getUpperBound(), eReference.getName());
+		Optional<String> source = modelState.getIndex().getSemanticId(eReference.getEContainingClass());
+		Optional<String> target = modelState.getIndex().getSemanticId(eReference.getEReferenceType());
+		if (source.isEmpty() || target.isEmpty()) {
+			return null;
+		}
+		String id = toId(eReference);
+		return new GEdgeBuilder().type(eReference.isContainment() ? Types.COMPOSITION : Types.REFERENCE) //
+				.id(id) //
+				.addCssClass(CSS.ECORE_EDGE) //
+				.addCssClass(eReference.isContainment() ? CSS.COMPOSITION : null) //
+				.add(new GLabelBuilder(Types.LABEL_EDGE) //
+						.edgePlacement(new GEdgePlacementBuilder()//
+								.side(GConstants.EdgeSide.TOP)//
+								.position(0.5d)//
+								.offset(0) //
+								.rotate(false) //
+								.build())//
+						.id(id + "_label") //
+						.text(label).build())
+				.sourceId(source.get()) //
+				.targetId(target.get()) //
+				.routerKind(GConstants.RouterKind.MANHATTAN)//
+				.build();
+	}
+
+	public GEdge create(EClass baseClass, EClass superClass) {
+		Optional<String> sourceId = modelState.getIndex().getSemanticId(baseClass);
+		Optional<String> targetId = modelState.getIndex().getSemanticId(superClass);
+		if (sourceId.isEmpty() || sourceId.isEmpty()) {
```

Comment:
Should be `|| targetId.isEmpty()` (But in general, returning null here will just cause another exception, because then you'd be trying to add a `null` edge to the EMF Model, which is forbidden)